components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/APIMappingUtil.java

```diff
@@ -2089,6 +2089,24 @@ private static APIOperationsDTO getOperationFromURITemplate(URITemplate uriTempl
             supportedMethods = APIConstants.HTTP_DEFAULT_METHODS;
         }
 
+        String defaultThrottlingPolicy = APIConstants.UNLIMITED_TIER;
+        if (!APIUtil.isEnabledUnlimitedTier()) {
+            // Set an available value if the Unlimited policy is disabled
+            String tenantDomain = RestApiCommonUtil.getLoggedInUserTenantDomain();
+            try {
+                Map<String, Tier> tierMap = APIUtil.getTiers(APIConstants.TIER_RESOURCE_TYPE, tenantDomain);
+                if (tierMap.size() > 0) {
+                    defaultThrottlingPolicy = tierMap.keySet().toArray()[0].toString();
+                } else {
+                    log.error("No throttle policies available in the tenant " + tenantDomain);
+                }
+            } catch (APIManagementException e) {
+                log.error("Error while getting throttle policies for tenant " + tenantDomain);
+            }
+
+        }
```

Comment:
shall we move this entire block to APIUtil with a method and call it from here?

Reply:
eg: APIUtil.getDefaultThrottlingPolicy(tenantDomain) -> returns Unlimited or a correct one available.

Reply:
Updated with 6c7341a